### PR TITLE
Update Finch to 0.11-M2

### PR DESCRIPTION
**Pull request by vkostyukov**
This RB updates Finch to 0.11-M2 as well as changes the implementation to reflect its best practices.

### CODE REVIEW

File: frameworks/Scala/finch/src/main/scala/Main.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.twitter.util.Await
+
+import io.circe.Json
+import io.finch._
+import io.finch.circe._
+
+object Main extends App {
+
+  val dateFormat: ThreadLocal[DateFormat] = new ThreadLocal[DateFormat] {
+    override def initialValue: DateFormat = new SimpleDateFormat("E, dd MMM yyyy HH:mm:ss z")
```
Comment (martin-g) [inline frameworks/Scala/finch/src/main/scala/Main.scala]: Why don't you use Java 8 DateTimeFormatter instead ?
It is thread-safe, so you'll need just one instance.

Comment (vkostyukov) [inline frameworks/Scala/finch/src/main/scala/Main.scala]: Good point! Thanks!

Let me update the PR.

**Comment by knewmanTE** [comment]
For some reason Travis fired off 2 runs for your pull request, so I cancelled [the one that seemed to be a bug](https://travis-ci.org/TechEmpower/FrameworkBenchmarks/builds/154543626). The other Travis run is still scheduled to run and can be followed [here](https://travis-ci.org/TechEmpower/FrameworkBenchmarks/builds/154580050) (or by clicking the yellow circle next to your commit above).

**Comment by vkostyukov** [comment]
Somehow this failed on the Erlang test. @knewmanTE do you want me to restart the build?

**Comment by knewmanTE** [comment]
@vkostyukov don't worry about the Erlang test. Misultin has been removed from the benchmark tests, but your Travis run thinks it still exists, so it's trying to check the directory but not finding anything there. Won't be an issue on master.

**Pull request merged**